src/activity_handlers.cpp

```diff
@@ -1501,6 +1502,51 @@ void activity_handlers::repair_item_finish( player_activity *act, player *p )
     act->moves_left = actor->move_cost;
 }
 
+void activity_handlers::mend_item_finish( player_activity *act, player *p )
+{
+    item_location target;
+    switch( static_cast<item_location::type>( act->index ) ) {
```

Comment:
The more I think about it the worse it sounds.

Could we just reuse the idea from crafting for now? That is, instead of tracking item location, just consume the first-best item that solves the problem.
It will stop being correct when replacement parts become actual parts of the items, but it will be a while until that happens.